lib/chef/resource/ohai_hint.rb

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright:: Copyright 2011-2018, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class OhaiHint < Chef::Resource
+      resource_name :ohai_hint
+      provides :ohai_hint
+
+      description "Use the ohai_hint resource to pass hint data to Ohai to aid in configuration detection."
+      introduced "14.0"
+
+      property :hint_name,
+               String,
+               description: "The name of hints file if different from the resource name",
+               name_property: true
+
+      property :content,
+               Hash,
+               description: "A Hash of values to include in the hint file"
```

Comment:
> Data to include in the hint file

We already know it's a hash, and our docs should show the type and the description.